ledger/src/shred/merkle.rs

```diff
@@ -1178,7 +1178,8 @@ pub(super) fn make_shreds_from_data(
     };
     // Split the data into erasure batches and initialize
     // data and coding shreds for each batch.
-    while data.len() >= 2 * chunk_size || data.len() == chunk_size {
+    while data.len() >= chunk_size {
```

Comment:
Imagine you have 321 shreds worth of data, `is_last_in_slot = true`.

The old code will make 9 batches of 32:32 shreds (288) and 1 batch of 33:33 shreds.

This code we will instead make 10 batches of 32:32 shreds (320) and 1 batch of 32:32 shreds where 31 of those shreds are padding.

Reply:
well yes, but the reality is such that the code typically deals with e.g. 55 shreds worth of data, and ends up making a 59:59 batches. This consistently overstresses RSC decoder. A better solution to your scenario would be to make 2 batches of 16:16 and 17:16 respectively.

Reply:
the very last batch in a slot needs to be `32:32 + `. 

To be clear I'm fine with requiring that each batch is <= 32:32, and increasing our target later. 
I'm just wondering if we can avoid unnecessary padding for the last set. So if `is_last_in_slot = true`, allow an exemption for the very last batch to be greater than 32:32.

Or we could add the small set in the middle if we need to be firm on the 32 max size.
Using the example above we make 9 batches of 32:32 (288), one batch of 1:17 (289), and finally one batch of 32:32

Reply:
My understanding from @lidatong is that they highly desire batches that are exactly 32:32 in size so that they can hard code this assumption into other areas of the code and simplify a bunch of logic

Reply:
Yes 32:32 would simplify the logic responsible for figuring out in which coding batch a given shred would land (as right now it calls for something like a hashmap with FEC Set Index as key, which is slow) and fixed FEC set size would allow a simple vector to be used instead. This in turn simplifies signaling for repair etc etc. I think we can try hardwiring 32:32 and measure if it actually bad at all on testnet. Its minimal effort and easy enough to observe how many zeros we ultimatly end up shipping on the wire.

Reply:
yep, we're aligned